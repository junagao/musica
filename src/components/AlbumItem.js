import React from 'react';
import PropTypes from 'prop-types';

import './AlbumItem.scss';

const AlbumItem = ({ artist, artwork, album }) => (
  <li className="album-item">
    <img src={artwork.replace('100x100', '200x200')} alt="" />
    <div className="album-details">
      <p className="album-title">{album}</p>
      <p className="album-artist">{artist}</p>
    </div>
  </li>
);

AlbumItem.propTypes = {
  artist: PropTypes.string.isRequired,
  artwork: PropTypes.string.isRequired,
  album: PropTypes.string.isRequired,
};

export default AlbumItem;
