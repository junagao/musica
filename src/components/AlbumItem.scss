@import '../variables';

.album-item {
  border-radius: 0.25rem;
  border: 1px solid transparent;
  width: 150px;
  animation: fadeIn 1s;
}

img {
  border-radius: 0.25rem;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  width: 100%;
}

.album-details {
  padding: 0.25rem 0rem;
}

.album-title {
  color: $color-album-title;
  font-size: 0.95rem;
  font-weight: 600;
}

.album-artist {
  color: $color-album-artist;
  font-size: 0.85rem;
  font-weight: 400;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
