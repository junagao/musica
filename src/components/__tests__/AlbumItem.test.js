import React from 'react';
import { shallow } from 'enzyme';

import AlbumItem from 'components/AlbumItem';

describe('AlbumItem', () => {
  let wrapper;
  const artist = 'Tycho';
  const artwork =
    'https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/200x200bb.jpg';
  const album = 'Awake';

  beforeEach(() => {
    wrapper = shallow(
      <AlbumItem artist={artist} artwork={artwork} album={album} />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one album per li item with className of album-item', () => {
    expect(wrapper.find('li').length).toEqual(1);
    expect(wrapper.find('li').hasClass('album-item')).toEqual(true);
  });

  it('should render the album image correctly', () => {
    expect(wrapper.find('img').length).toEqual(1);
    expect(wrapper.find('img').prop('src')).toEqual(artwork);
  });

  it('should render one div with className of album-details', () => {
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('div').hasClass('album-details')).toEqual(true);
  });

  it('should render two p elements', () => {
    expect(wrapper.find('p').length).toEqual(2);
  });

  it('should render one p with className album-title', () => {
    expect(wrapper.find('p').at(0).length).toEqual(1);

    expect(
      wrapper
        .find('p')
        .at(0)
        .hasClass('album-title'),
    ).toEqual(true);
  });

  it('should render the album title correctly', () => {
    expect(
      wrapper
        .find('p')
        .at(0)
        .text(),
    ).toEqual(album);
  });

  it('should render one p with className album-artist', () => {
    expect(wrapper.find('p').at(1).length).toEqual(1);

    expect(
      wrapper
        .find('p')
        .at(1)
        .hasClass('album-artist'),
    ).toEqual(true);
  });

  it('should render the album artist correctly', () => {
    expect(
      wrapper
        .find('p')
        .at(1)
        .text(),
    ).toEqual(artist);
  });
});
