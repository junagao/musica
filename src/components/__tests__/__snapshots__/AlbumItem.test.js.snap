// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumItem should render correctly given the required props 1`] = `
<li
  className="album-item grid"
>
  <img
    alt=""
    src="https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/200x200bb.jpg"
  />
  <div
    className="album-details"
  >
    <p
      className="album-title"
    >
      Awake
    </p>
    <p
      className="album-artist"
    >
      Tycho
    </p>
  </div>
</li>
`;
